lib/rules/no-expose-after-await.js

```diff
@@ -40,8 +40,7 @@ module.exports = {
     type: 'problem',
     docs: {
       description: 'disallow asynchronously registered `expose`',
-      categories: undefined,
-      // categories: ['vue3-essential'], TODO Change with the major version
+      categories: ['vue3-essential'],
```

Comment:
This rule (and some others) work fine on Vue 2 with [`@vue/composition-api`](https://github.com/vuejs/composition-api). Should we also add them to the respective Vue 2 presets, to help smooth the transition to Vue 3?

Or maybe create a new preset for those rules only that can be applied on top of any Vue 2 preset?

Reply:
I'm not sure about `@vue/composition-api` because I'm not using `@vue/composition-api`. 
I think it's okay if we add a rule to the preset if you think users need it.
However, I've heard before that `@vue/composition-api` doesn't support async, so if that's correct, I think the `no-expose-after-await` rule should only be included in the preset for Vue3.

Reply:
You're right, `@vue/composition-api` doesn't support async `setup`, so this rule (and some others) are not required on Vue 2. However, there are some rules from which Vue 2 users would profit:

#### For [`@vue/composition-api`](https://github.com/vuejs/composition-api) users

* `vue/no-setup-props-destructure`
* `vue/no-ref-as-operand`

#### For https://github.com/antfu/unplugin-vue2-script-setup/ users

* `vue/no-export-in-script-setup`
* `vue/return-in-emits-validator`
* `vue/valid-define-emits`
* `vue/valid-define-props`

Reply:
Thank you for listing the rules!
I will include them in the preset for vue2.